### PR TITLE
Labproject/clone template

### LINKED ISSUE #13: Initialise repository based on template with optional submodules
The research project management service (and in fact this whole library) was created because of the need to simplify creating research project repositories based on templates with submodules.

### PR DESCRIPTION
**Pull request by achilleas-k**
This PR introduces the ability for the labproject service to create a new project repository by cloning a template repository with submodules.  All submodules are cloned and pushed to the server as well as the main project.  All new repositories are added to the project team.

These changes required augmenting the tonic Client type to hold a GINClient reference for calling gin-cli code for cloning and running git commands.

Full documentation for setting up and running the service with the new features will come in a followup PR.

Closes #13

### CODE REVIEW

File: utonics/labproject/main.go [hunk 1]
```diff
@@ -272,19 +400,111 @@ func getAvailableOrgsAndTeams(botClient, userClient *worker.Client) (map[string]
...
 	}
-	return userpass["username"], userpass["password"]
+
+	if config.Config == nil {
+		config.Config = new(tonic.Config)
+	}
+
+	// Set defaults for any unset values
+	if config.GIN.Web == "" {
+		config.GIN.Web = "https://gin.dev.g-node.org:443"
```
Comment (mpsonntag) [inline utonics/labproject/main.go:424]: Should we add a ToDo here to set an appropriate default value after the development stage?

File: utonics/labproject/main.go [hunk 2]
```diff
@@ -272,19 +400,111 @@ func getAvailableOrgsAndTeams(botClient, userClient *worker.Client) (map[string]
...
+		log.Printf("[config] Setting default GIN server web address: %s", config.GIN.Web)
+	}
+	if config.GIN.Git == "" {
+		config.GIN.Git = "git@gin.dev.g-node.org:2424"
```
Comment (mpsonntag) [inline utonics/labproject/main.go:428]: Same comment as above.

**Review by mpsonntag — APPROVED** [review]
LGTM

File: utonics/labproject/main.go (hunk 1 above)
Comment (achilleas-k) [reply utonics/labproject/main.go:424]: Hm.  You know, it's probably best to make it fail if it's not set.  The address the service talks to is important enough to not rely on an implied default.  This service isn't expected to run against the main GIN service, so we wouldn't want to add that one like we would with something like Valid or DOI.

Let me update this.

**Pull request merged**